app/src/main/java/com/mtech/parttimeone/photolearn/ViewModel/LearningSessionViewModel.java

```diff
@@ -174,11 +175,35 @@ public void deleteLearningSession(String sessionId, String userId) {
     // TODO
     // This is for trainers to update a learning session
     public LearningSessionBO updateLearningSession(LearningSessionBO learningSessionBO, String sessionId , String userId) {
-        mLearningSession = FirebaseDatabase.getInstance().getReference(learningSessionRepository.getRootNode());
-
         //since the actual changes are not specified so update entire node
+
+        //update main session data
+        mLearningSession = FirebaseDatabase.getInstance().getReference(learningSessionRepository.getRootNode());
         mLearningSession.child(sessionId).setValue(learningSessionBO);
 
+        //update trainer session data
+        mUserLearningSession = FirebaseDatabase.getInstance().getReference(userLearningSessionRepository.getRootNode());
+        mUserLearningSession.child("trainers").child(userId).child(sessionId).setValue(learningSessionBO);
+
+       //update participant session data
+        mUserLearningSession = FirebaseDatabase.getInstance().getReference(userLearningSessionRepository.getRootNode());
+        mUserLearningSession.child("participants").addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot dataSnapshot) {
+                if (dataSnapshot.getValue() != null) {
+                    for (DataSnapshot sessionSnapshot : dataSnapshot.getChildren()) {
+                        if(sessionSnapshot.hasChild(sessionId)) {
+                            sessionSnapshot.child(sessionId).getRef().setValue(learningSessionBO);
```

Comment:
can you use the update method instead?

Reply:
https://firebase.google.com/docs/database/android/read-and-write

see if the updateChildren method can be used instead